test/e2e/mob_test.go

```diff
@@ -0,0 +1,46 @@
+package e2e_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestOnClusterBuildLabel(t *testing.T) {
```

Comment:
What is the intent of this test? 

It looks like it: 
- creates an MCP
- labels a node with the on-cluster-build role
- creates a machineconfig 
- then checks to make sure the node had the role label? 

(it doesn't look like it checks for `ctrlcommon.LayeringEnabledPoolLabel` or see if the MOB deployment scaled or not, and I'd maybe expect it to if it was testing the code in this PR? This might just be "not done yet, go away @jkyros " :smile: )